app/src/main/java/com/kamwithk/ankiconnectandroid/ankidroid_api/IntegratedAPI.java

```diff
@@ -78,6 +78,55 @@ public Long addNote(final Map<String, String> data, String deck_name, String mod
         }
     }
 
+    public boolean updateNoteFields(long note_id, Map<String, String> new_fields, final Map<String, byte[]> files_to_add, final Map<String, ArrayList<String>> field_to_filenames_img, final Map<String, ArrayList<String>> field_to_filenames_sound) throws Exception {
+        String[] model_field_names = modelAPI.modelFieldNames(noteAPI.getNoteModelId(note_id));
+        String[] card_fields = noteAPI.getNoteFields(note_id);
+        Map<String, String> stored_files = new HashMap<>();
```

Comment:
It seems like `stored_files` only exists to check for duplicate files. I don't think Anki-Connect handles duplicate files unless specified with an additional [`deleteExisting`](https://github.com/FooSoft/anki-connect/blob/bf5c73b9cdec95337dc356b0d006b7847458c086/plugin/__init__.py#L673) field, so this logic can be safely removed.

Reply:
Actually the map is there because `mediaAPI.storeMediaFile()` doesn't produce the same filename as what's fed into it (at least when I was testing I think), so I cached the filename from the first time `mediaAPI.storeMediaFile()` was called. Otherwise, you'd store the same file multiple times within the same `updateNoteFields()` call under different filenames. More elegant solution would be to find out why `mediaAPI.storeMediaFile()` doesn't store with the exact filename it's called with, but that's why this map exists at least.

Reply:
Oh interesting, thanks for the clarification! In this case, would it be possible to put a comment explaining exactly that (and when the time comes, add that to the documentation too)?

Reply:
Yeah can do.